src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,194 @@
+import classNames from 'classnames';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { changedTodo, deleteTodo } from '../../api/todos';
+import { ErrorType } from '../../types/ErrorType';
+import { Todo } from '../../types/Todo';
+import { ContextTextError } from '../Context/ContextTextError';
+import { ContextTodos } from '../Context/ContextTodos';
+import { ContextToggleAll } from '../Context/ContextToggleAll';
+import { TodoLoader } from '../TodoLoader';
+
+import './TodoItem.scss';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({ todo }) => {
+  const { title, completed, id } = todo;
+
+  const { setTextError } = useContext(ContextTextError);
+  const { todos, setTodos, isAddingTodo } = useContext(ContextTodos);
+  const {
+    isToggleAllCompleted,
+    isToggleAllUnCompleted,
+  } = useContext(ContextToggleAll);
+
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [newTitle, setNewTitle] = useState<string>(title);
+
+  const newTitleField = useRef<HTMLInputElement>(null);
+
+  const unCompletedTodosId = todos.filter((el) => !el.completed)
+    .map((el) => el.id);
+  const completedTodosId = todos.filter((el) => el.completed)
+    .map((el) => el.id);
+  const conditionalLoading = (isLoading && (selectedTodoId === id))
+  || (isToggleAllUnCompleted && unCompletedTodosId.includes(id))
+  || (isToggleAllCompleted && completedTodosId.includes(id))
+  || (isAddingTodo && id === 0);
+
+  const deleteTodoOnServer = async () => {
+    try {
+      await deleteTodo(id);
+      setTodos(curentTodos => [...curentTodos].filter(el => el.id !== id));
+    } catch {
+      setTextError(ErrorType.DELETE);
+    } finally {
+      setSelectedTodoId(null);
+      setIsLoading(false);
+    }
+  };
+
+  const changedTodoOnServer = async (
+    updateTitle: string | null,
+    updateCompleted: boolean | null,
+    todoId: number,
+  ) => {
+    try {
+      await changedTodo(todoId, updateTitle, updateCompleted);
+      setTodos(curentTodos => [...curentTodos].map(el => {
+        if (el.id === todoId) {
+          if (updateCompleted !== null) {
+            return { ...el, completed: !el.completed };
+          }
+
+          if (updateTitle) {
+            return { ...el, title: updateTitle };
+          }
+        }
+
+        return el;
+      }));
+    } catch (error) {
+      setTextError(ErrorType.PATCH);
+    }
+  };
+
+  const handlerDeleteTodo = () => {
+    setIsLoading(true);
+    setSelectedTodoId(id);
+    deleteTodoOnServer();
+  };
+
+  const handlerToggleClick = () => {
+    setIsLoading(true);
+    setSelectedTodoId(id);
+    changedTodoOnServer(null, !completed, id)
+      .finally(() => {
```

Comment:
You're already using one construction, so use it everywhere
![image](https://user-images.githubusercontent.com/70009205/211387007-5ee9322f-b6b4-4bea-bca1-d06551ca93ca.png)